mm/ktsan/memblock.c

```diff
@@ -17,14 +18,19 @@ uptr_t kt_memblock_addr(uptr_t addr)
 	if (!virt_addr_valid(addr))
 		return 0;
 	page = virt_to_head_page((void *)addr);
-	if (!PageSlab(page))
-		return 0;
-	cache = page->slab_cache;
-	offset = addr - (uptr_t)page->s_mem;
-	idx = reciprocal_divide(offset, cache->reciprocal_buffer_size);
-	obj_addr = (uptr_t)(page->s_mem + cache->size * idx);
 
-	return obj_addr;
+	/* If the page is a slab page, we want to delete
+	   sync objects when a slab object is freed. */
+	if (PageSlab(page)) {
+		cache = page->slab_cache;
+		offset = addr - (uptr_t)page->s_mem;
+		idx = reciprocal_divide(offset, cache->reciprocal_buffer_size);
+		obj_addr = (uptr_t)(page->s_mem + cache->size * idx);
+		return obj_addr;
+	}
+
+	/* Add 1 to avoid collisions with the first object on a slab page. */
+	return (uptr_t)page_address(page) + 1;
```

Comment:
This is not a slab page, right? We've handled slab pages above. So I don't understand the +1 logic. Why exactly do we need +1?